File: application/app/src/main/java/com/yahyeet/boardbook/model/entity/AbstractEntity.java
package com.yahyeet.boardbook.model.entity;

import androidx.annotation.Nullable;

public abstract class AbstractEntity {
	private String id;

	public AbstractEntity(String id) {
		this.id = id;
	}

	public AbstractEntity() {
	}

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	@Override
	public boolean equals(@Nullable Object obj) {
		if (obj == this) {
			return true;
		}

		if (!(obj instanceof AbstractEntity)) {
			return false;
		}

		AbstractEntity entity = (AbstractEntity) obj;
		return id.equals(entity.id);
	}
}

File: application/app/src/main/java/com/yahyeet/boardbook/model/entity/AbstractEntity.kt
package com.yahyeet.boardbook.model.entity

abstract class AbstractEntity {
    var id: String? = null

    constructor(id: String) {
        this.id = id
    }

    constructor() {}

    override fun equals(obj: Any?): Boolean {
        if (obj === this) {
            return true
        }

        if (obj !is AbstractEntity) {
            return false
        }

        val entity = obj as AbstractEntity?
        return id == entity!!.id
    }
}
